pkg/resources/kafka/kafka.go

```diff
@@ -240,6 +240,7 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 	}
 
 	reorderedBrokers := reorderBrokers(brokerPods, r.KafkaCluster.Spec.Brokers, r.KafkaCluster.Status.BrokersState, controllerID)
+	isPerBrokerDynamicConfigFailed := false
```

Comment:
How about `allBrokerDynamicConfigSucceeded := true` and than if an error occurs set it to `false`